sdk/samples/core/paho_telemetry_producer_sample.c

```diff
@@ -0,0 +1,206 @@
+  }
+
+  return AZ_OK;
+}
+
+az_result telemetry_send_begin(az_span send_telemetry_name, az_span payload, int8_t qos)
+{
+  az_mqtt5_telemetry_producer_send_req_event_data telemetry_data
+      = { .content_type = content_type,
+          .telemetry_name = send_telemetry_name,
+          .telemetry_payload = payload,
+          .qos = qos };
+  az_result rc = az_mqtt5_telemetry_producer_send_begin(&telemetry_producer, &telemetry_data);
+  if (az_result_failed(rc))
+  {
+    printf(
+        LOG_APP_ERROR "Failed to send telemetry '%s' with rc: %s\n",
+        az_span_ptr(send_telemetry_name),
+        az_result_to_string(rc));
+  }
+  return AZ_OK;
+}
+
+int main(int argc, char* argv[])
+{
+  (void)argc;
+  (void)argv;
+
+  az_log_set_message_callback(az_sdk_log_callback);
+  az_log_set_classification_filter_callback(az_sdk_log_filter_callback);
+
+  connection_context = az_context_create_with_expiration(
+      &az_context_application, az_context_get_expiration(&az_context_application));
+
+  az_mqtt5 mqtt5;
+  MQTTAsync mqtt_handle;
+
+  LOG_AND_EXIT_IF_FAILED(az_mqtt5_init(&mqtt5, &mqtt_handle, NULL));
+
+  az_mqtt5_x509_client_certificate primary_credential = (az_mqtt5_x509_client_certificate){
+    .cert = cert_path1,
+    .key = key_path1,
+  };
+
+  az_mqtt5_connection_options connection_options = az_mqtt5_connection_options_default();
+  connection_options.client_id_buffer = client_id;
+  connection_options.username_buffer = username;
+  connection_options.password_buffer = AZ_SPAN_EMPTY;
+  connection_options.hostname = hostname;
+  connection_options.client_certificates[0] = primary_credential;
+  connection_options.client_certificate_count = 1;
+
+  LOG_AND_EXIT_IF_FAILED(az_mqtt5_connection_init(
+      &mqtt_connection, &connection_context, &mqtt5, mqtt_callback, &connection_options, NULL));
+
+  az_mqtt5_property_bag property_bag;
+  MQTTProperties prop = MQTTProperties_initializer;
+  LOG_AND_EXIT_IF_FAILED(az_mqtt5_property_bag_init(&property_bag, &mqtt5, &prop));
+
+  az_mqtt5_telemetry_producer_codec_options telemetry_producer_codec_options
+      = az_mqtt5_telemetry_producer_codec_options_default();
+
+  LOG_AND_EXIT_IF_FAILED(az_mqtt5_telemetry_producer_init(
+      &telemetry_producer,
+      &telemetry_producer_codec,
+      &mqtt_connection,
+      property_bag,
+      client_id,
+      model_id,
+      AZ_SPAN_FROM_BUFFER(telemetry_topic_buffer),
+      AZ_MQTT5_TELEMETRY_DEFAULT_TIMEOUT_SECONDS,
+      &telemetry_producer_codec_options));
+
+  LOG_AND_EXIT_IF_FAILED(az_mqtt5_connection_open(&mqtt_connection));
+
+  // infinite execution loop
+  for (int i = 0; !sample_finished; i++)
+  {
+    printf(LOG_APP "Waiting...\r");
+    fflush(stdout);
+
+    // sends a telemetry message every 15 seconds. This cadence/how it is triggered should be
+    // customized for your solution.
+    if (i % 15 == 0)
+    {
+      // TODO: Payload should be generated and serialized
+      LOG_AND_EXIT_IF_FAILED(telemetry_send_begin(
```

Comment:
this always returns AZ_OK

Reply:
yep - log and exit if failed should only be triggered here on an unrecoverable error